rest/utilities_testing.go

```diff
@@ -236,10 +268,16 @@ func (rt *RestTester) Bucket() base.Bucket {
 			// If no db config was passed in, create one
 			rt.DatabaseConfig = &DatabaseConfig{}
 		}
-
 		if rt.DatabaseConfig.UseViews == nil {
 			rt.DatabaseConfig.UseViews = base.BoolPtr(base.TestsDisableGSI())
 		}
+		if rt.collectionConfig != useSingleCollectionDefaultOnly && (!*rt.DatabaseConfig.UseViews || base.UnitTestUrlIsWalrus()) {
```

Comment:
This condition doesn't handle the case where rt.collectionConfig = useSingleCollection and SG_TEST_USE_DEFAULT_COLLECTION=true.  I'm not clear why that isn't needed.